### PR TITLE
TESTS: Remove Dead Code in Test Infra.

### PR DESCRIPTION
**Pull request by original-brownbear**
* None of this infrastructure is used
* Some redundant throws and resulting catch code removed

### CODE REVIEW

**Comment by original-brownbear** [comment]
Jenkins test this

**Review by alpar-t — APPROVED** [review]
LGTM

File: test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java [hunk 1]
```diff
@@ -183,7 +182,6 @@
 import java.util.concurrent.Callable;
 import java.util.concurrent.CopyOnWriteArrayList;
 import java.util.concurrent.CountDownLatch;
-import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/ESIntegTestCase.java]: I :heart: anything that makes this class smaller.

File: test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java [hunk 2]
```diff
@@ -1718,13 +1682,6 @@ public void fullRestart() throws Exception {
         fullRestart(EMPTY_CALLBACK);
     }
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: It'd be neat to look at usages of `fullRestart` and see if we've implemented all of them in our full cluster restart qa tests.

Comment (nik9000) [reply test/framework/src/main/java/org/elasticsearch/test/InternalTestCluster.java]: Like, later. Not as part of this.

**Review by nik9000 — APPROVED** [review]

**Comment by original-brownbear** [comment]
@nik9000 @atorok thanks for the review!

**Pull request merged**